src/components/v5/common/ActionSidebar/partials/TokensTable/hooks.tsx

```diff
@@ -23,48 +30,112 @@ export const useTokensTableColumns = (
       columnHelper.display({
         id: 'token',
         header: () => formatText({ id: 'table.row.token' }),
-        cell: ({ row }) => <TokenSelect name={`${name}.${row.index}.token`} />,
+        cell: ({ row }) => (
+          <div className="w-full py-2 sm:py-0">
+            <TokenSelect name={`${name}.${row.index}.token`} />
+          </div>
+        ),
       }),
       columnHelper.display({
         id: 'symbol',
         header: () => formatText({ id: 'table.row.symbol' }),
-        cell: ({ row }) => (
-          <TokenSymbol address={dataRef.current?.[row.index]?.token} />
-        ),
+        cell: ({ row }) => {
+          const showNewLabel = dataRef.current?.[row.index]?.isNew;
+          const showRemovedLabel = !!dataRef.current?.[row.index]?.isRemoved;
+
+          const tokenSymbol = (
+            <div className="w-full py-2 sm:py-0">
+              <TokenSymbol address={dataRef.current?.[row.index]?.token} />
+            </div>
+          );
+
+          return showNewLabel || showRemovedLabel ? (
+            <div className="w-full flex flex-col items-start gap-2 sm:gap-4 sm:flex-row sm:items-center sm:justify-between">
+              {tokenSymbol}
+              <PillsBase
+                className={clsx('sm:ml-auto', {
+                  'text-negative-400 bg-negative-100': showRemovedLabel,
+                  'text-success-400 bg-success-100': showNewLabel,
+                })}
+                text={formatText({
+                  id: showRemovedLabel ? 'badge.removed' : 'badge.new',
+                })}
+              />
+            </div>
+          ) : (
+            tokenSymbol
+          );
+        },
       }),
     ],
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [columnHelper, name],
+    [columnHelper, dataRef, name],
   );
 
   return columns;
 };
 
 export const useGetTableMenuProps = (
-  { remove },
+  { update, remove },
   data,
   shouldShowMenu: Exclude<TokensTableProps['shouldShowMenu'], undefined>,
 ) => {
+  const { colony } = useColonyContext();
+
+  const colonyTokens = useMemo(
+    () => colony?.tokens?.items.filter(notNull) || [],
+    [colony?.tokens?.items],
+  );
+
   return useCallback<
     TableWithMeatballMenuProps<TokensTableModel>['getMenuProps']
   >(
```

Comment:
Due to the nature of this, it will be recreated on each re-render since both the dependencies and the arguments have a high likelihood of changing.

This means, that use callback cannot memoize the function's instance, and will recreate it each time. So using it here, actually hurts performance, due to it's overhead.